### PR TITLE
feat: Add FieldState.loading.

### PR DESCRIPTION
**Pull request by stephenh**
Currently it will be set for free from `useFormState` that does:

```
const fs = useFormState({ config, init: { data: ..., map: ... } });
```

When `data` is undefined. This should work for our existing usages that typically do:

```
const fs = useFormState({ config, init: { data: query.data?.someFoo, map: ... } });
```

But I'll work on a follow up PR that allows passing the `query` in out-right, i.e.:

```
const query = useFoo();
const fs = useFormState({ config, init: { query, map: ... } });
```

### CODE REVIEW

File: src/fields/valueField.ts [hunk 1]
```diff
@@ -123,9 +124,19 @@ export function newValueFieldState<T, K extends keyof T>(
       return parentState().readOnly || this._readOnly;
     },
 
-    /** Sets the field readOnly, but `v` will still be or-d with the parent's readOnly state. */
-    set readOnly(v: boolean) {
-      this._readOnly = v;
+    /** Sets the field readOnly, but `loading` will still be or-d with the parent's readOnly state. */
+    set readOnly(loading: boolean) {
```
Comment (bdow) [inline src/fields/valueField.ts:128]: I kind of find it odd that the parameter's name is `loading`. Seems to couple the readOnly state with the loading state, where I'd think `readOnly` can be `true`, even if `loading === false`.

Comment (bdow) [reply src/fields/valueField.ts:128]: was this on purpose?

**Comment by TylerR909** [comment]
I'd be wary of this `data === undefined ---> loading` paradigm. Say we update forms to have Skeletons or similar loading state, and our Query errored out. Just because data is undefined doesn't mean we should be signaling to the user that something is still in-flight.

```js
const {
  data, // undefined
  error, // {...}
  loading, // false
} = useSomeBrokenQuery();
```

I'd hope for a toast notification or some other Error state, but the Form should not indicate anything is loading at this point.

I like "Clever" code so long as it's not "Too clever by half," and data->undefined->Loading!! crosses that line imo.

Even making forms aware of GQL Queries seems precisely 1 bridge too far. i.e. why not just...

```js
const { loading, data } = useFoo();
const fs = useFormState({ config, loading, init: { data, map: ... } });
```

**Comment by stephenh** [comment]
```
const fs = useFormState({ config, loading, init: { data, map: ... } });
```

Fwiw I do like renaming the `init.input` key to `input.data` so that that destructure-d `data` key lines up for `init`'s input...

**Review by bdow — APPROVED** [review]

**Comment by stephenh** [comment]
@TylerR909 was chatting with @bdow , I'd like to just try this for now with the rationale of:

a) nothing in the app actually uses loaded, so it won't immediately break/change anything,
b) the admitted heuristic of `data === undefined` --> `loading` gives all of our existing forms the notion of loading for us to experiment with
c) I'm going to do a follow up PR that allows passing in `query` / `query.loading` directly, b/c I do think that pattern is a good idea
d) I/we definitely reserve the right to revert/change any of ^ if the heuristic really is too naive/not worthwhile :-) 

Basically just want to experiment and see what happens. I think it's maybe 50/50 that we rollback the `data === undefined` guess...
